### ISSUE TITLE
Possible errors in mask_embeds in incremental learning

### ISSUE THREAD

**Issue by hanminpay** [clovaai/ECLIPSE#26]
Hello, great work on this project! I was reading through the code and found a potential issue that I'm not sure if it's intentional or a bug.

In the file `mask2former/modeling/transformer_decoder/mask2former_transformer_decoder.py` at **line 644**, the following code snippet appears:

```python
if self.num_prompts > 0:
    mask_embeds = nn.ModuleList([self.mask_embed])
    mask_embeds = mask_embeds.extend(self.prompt_mask_embed)
    query_dims = np.cumsum([0, self.num_queries] + [self.num_prompts] * len(self.prompt_embed))
else:
    mask_embeds = self.mask_embed
    query_dims = None
```

I noticed that `mask_embeds = mask_embeds.extend(self.prompt_mask_embed)` may be problematic. The `extend()` method of `nn.ModuleList` (like Python's built-in `list.extend()`) modifies the list in-place and **returns `None`**. This means `mask_embeds` will be assigned `None` after this line, which could lead to errors in subsequent usage.

Should this instead be written as:
```python
mask_embeds.extend(self.prompt_mask_embed)
```
without reassigning the result?

I'd appreciate clarification on whether this is a bug or if the current behavior is intended. Thanks for your time!

**Comment by qjadud1994** [comment]
I appreciate your question about the implementation.

I checked the documentation of the `extend()` function of `nn.ModuleList` in [here](https://docs.pytorch.org/docs/stable/generated/torch.nn.ModuleList.html#torch.nn.ModuleList.extend).

I found that the `extend` function modifies the list in-place, as you mentioned, but does not return `None`; it returns `self` (modified module).

Therefore, both implementations are fine:
- `mask_embeds = mask_embeds.extend(self.prompt_mask_embed)`
- `mask_embeds.extend(self.prompt_mask_embed)`

I ran this simple script for testing:
~~~
import torch
import torch.nn as nn

module = nn.ModuleList([nn.ReLU()])
print(module)

module = module.extend([nn.GELU()])
print(module)

module.extend([nn.Sigmoid()])
print(module)
~~~

~~~
# output
ModuleList(
  (0): ReLU()
)
ModuleList(
  (0): ReLU()
  (1): GELU(approximate='none')
)
ModuleList(
  (0): ReLU()
  (1): GELU(approximate='none')
  (2): Sigmoid()
)
~~~

**Comment by hanminpay** [comment]
Thank you for your patience and detailed explanation! I apologize for the confusion — it was my mistake. I truly appreciate your time and the clear response. Wishing you all the best in your research and life!
